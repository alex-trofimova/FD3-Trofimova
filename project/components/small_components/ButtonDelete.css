.ButtonDelete_cross {
    cursor: pointer;
}

.ButtonDelete_cross:hover >
    .ButtonDelete_line {
        background-color: red;
        transition: 0.3s ease;   
    }

.ButtonDelete_line {
    position: absolute;
    width: 18px;
    height: 3px;
    
}

.cross_grey {
    background-color: rgb(241, 241, 241);
}

.cross_white {
    background-color: white;
} 

.ButtonDelete_first_line {
    transform: rotate(-45deg);
}
.ButtonDelete_second_line {
    transform: rotate(45deg);
}
