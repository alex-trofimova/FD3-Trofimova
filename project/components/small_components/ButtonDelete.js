import React from 'react';
import PropTypes from 'prop-types';

import './ButtonDelete.css';

class ButtonDelete extends React.PureComponent {

static propTypes = {
    classColor: PropTypes.string.isRequired,
};

  render() {
    
    return (
    <div className='ButtonDelete_cross' >
        <div className={this.props.classColor+' ButtonDelete_line ButtonDelete_first_line'}></div>
        <div className={this.props.classColor+' ButtonDelete_line ButtonDelete_second_line'}></div>
    </div>     
    )
    ;

  }
}

export default ButtonDelete;
