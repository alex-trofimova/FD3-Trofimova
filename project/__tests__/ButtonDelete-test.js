import React from 'react';
import renderer from 'react-test-renderer';

import ButtonDelete from '../components/small_components/ButtonDelete';

test('работа ButtonDelete', () => {

  const component = renderer.create(
    <ButtonDelete classColor='cross_white'/>
  );

  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
    
});
