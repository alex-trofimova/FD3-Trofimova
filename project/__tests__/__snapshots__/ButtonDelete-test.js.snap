// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа ButtonDelete 1`] = `
<div
  className="ButtonDelete_cross"
>
  <div
    className="cross_white ButtonDelete_line ButtonDelete_first_line"
  />
  <div
    className="cross_white ButtonDelete_line ButtonDelete_second_line"
  />
</div>
`;
